js/app/dashboard/store/reducers.js

```diff
@@ -0,0 +1,116 @@
+import { combineReducers } from 'redux'
+import _ from 'lodash'
+
+
+// initial data pumped from the template
+const tasksInitial = window.AppData.Dashboard.tasks,
+      unassignedTasksInitial = _.filter(tasksInitial, task => !task.isAssigned),
+      assignedTasksList = _.filter(tasksInitial, task => task.isAssigned),
+      assignedTasksByUserInitial = _.groupBy(assignedTasksList, task => task.assignedTo)
+
+/*
+  Store for all assigned tasks
+*/
+const assignedTasksByUser = (state = assignedTasksByUserInitial, action) => {
+
+  let  userTasks = state[action.username],
+    newState = { ...state },
+    newUserTasks,
+    taskComparator = (taskA, taskB) => taskA['@id'] === taskB['@id']
+
+  switch(action.type) {
+    case 'ASSIGN_TASKS':
+      newUserTasks = userTasks.slice()
+      newState[action.username] = Array.prototype.concat(newUserTasks, action.tasks)
+      newState[action.username].duration = 0
```

Comment:
Looks weird, is it an array or an object?

Reply:
A JS array is indeed an object `typeof([]) -> "object`. You can assign arbitrary attributes to it (here distance, duration) and use it as an Array at the same time. I think it does the job nicely here.

Reply:
Yes, but when you iterate over the values, how do you distinguish between tasks, and duration/distance?

Reply:
`newState[action.username]` **is** a javascript array so if you iterate over it you will iterate over the tasks. It has also all the array methods. It has the extra-attributes `distance` and `duration`.

In pseudo-PHP code it may look like this :
```php
class TaskArray extends Array {
     protected duration;
     protected distance;
}
```

Reply:
I didn't know this was possible 🙂
I think it looks more "natural" like this: 

```
newState[action.username].tasks = action.tasks
newState[action.username].duration = action.duration
newState[action.username].distance = action.distance
```